src/App.tsx

```diff
@@ -1,27 +1,68 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleClickAll = useCallback(() => {
+    const loadGoods = async () => {
+      const goodsData = await getAll();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      setGoods(goodsData);
+    };
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+    loadGoods();
+  }, []);
 
-    <GoodsList goods={[]} />
-  </div>
-);
+  const handleClickFirst5 = useCallback(() => {
+    const loadGoods = async () => {
+      const goodsData = await get5First();
+
+      setGoods(goodsData);
+    };
+
+    loadGoods();
+  }, []);
+
+  const handleClickReds = useCallback(() => {
+    const loadGoods = async () => {
+      const goodsData = await getRedGoods();
+
+      setGoods(goodsData);
+    };
+
+    loadGoods();
+  }, []);
```

Comment:
I'm pretty sure you can remove this one function creation, seems to be quite unnecessary here
```suggestion
  const handleClickReds = useCallback(async () => {
      const goodsData = await getRedGoods();

      setGoods(goodsData);
 
  }, []);
```